### ISSUE TITLE
[Tagging] Assign taxonomy to organizations

### ISSUE THREAD

**Issue by bradenmacdonald** [openedx/modular-learning#117]
"As a global staff member, I can assign taxonomies to organizations."

"As a global staff member, I can assign a taxonomy to all organizations."

"As a global staff member, I can assign a taxonomy to no particular organization."

Relates to https://github.com/openedx/modular-learning/issues/108 , https://github.com/openedx/modular-learning/issues/116 , and https://github.com/openedx/modular-learning/issues/109

### Acceptance Criteria

1. On the Taxonomy List Page, Indicate when a taxonomy is used by multiple organizations, as shown below.
1. For global staff users (and no other users), the "Manage organizations" menu will be added to the Taxonomies List Page and the Taxonomy Detail Page:
    ![Screenshot 2023-10-11 at 2 46 08 PM](https://github.com/openedx/modular-learning/assets/945577/d6c7203f-35b4-4ad3-a8c9-625ef6f7c230)
![Screenshot 2023-10-11 at 2 46 48 PM](https://github.com/openedx/modular-learning/assets/945577/a97fc774-8707-4c32-b509-2db2bfed1a3c)
2. This does not apply to system taxonomies.
3. Upon selecting that menu, this dialog will pop up, allowing the user to edit the assigned orgs:
    ![Screenshot 2023-10-11 at 2 50 07 PM](https://github.com/openedx/modular-learning/assets/945577/7a9ae99f-6feb-4951-9aa2-0ed8a1496eb4)
    ![Screenshot 2023-10-11 at 2 50 22 PM](https://github.com/openedx/modular-learning/assets/945577/f06cc5f9-7c70-4f7d-a62e-c3eed66c3976)
4. If no orgs are selected, display this warning:
    ![Screenshot 2023-10-11 at 2 51 19 PM](https://github.com/openedx/modular-learning/assets/945577/bba25593-1bbd-4cd7-a890-7593e5d6ad10)
4. Upon saving changes, a toast appears in the bottom left:
    ![Screenshot 2023-09-20 at 5 40 37 PM](https://github.com/openedx/modular-learning/assets/945577/9aa04892-78bb-4ed6-9e76-63e0edb63ff0)

### Developer Notes
* For **Criteria 3, the exact selection widget can change**. Try to find an existing widget that will do the job well, rather than implementing something custom. Someone had started [implementing a `Multiselect` component in Paragon](https://github.com/openedx/paragon/issues/1420) based on [React Select Multi](https://react-select.com/home), but the project has been on hold.
* Make sure the dialog code can be shared with https://github.com/openedx/modular-learning/issues/116.

**Comment by rpenido** [comment]
Hi @bradenmacdonald! CC @ali-hugo 
Do you think this is enough for the Autosuggest? I will fix the CSS to give more room for the dropdown.

![manage_orgs](https://github.com/openedx/modular-learning/assets/849463/4830c6c3-e9ce-48ae-a4e0-7d20607cb273)

It is not possible to clear the input the right way because the Autosuggest component does not forward the ref of the input (as other components usually do).

**Comment by bradenmacdonald** [comment]
@rpenido  That seems like a bug in paragon. The `Form.Autosuggest` should have higher priority than the modal it's in. Can you just keep the modal the same size but add a CSS rule so that the dropdown has a higher z-index?

> It is not possible to clear the input the right way because the Autosuggest component does not forward the ref of the input (as other components usually do).

What do you mean? You shouldn't need a ref to the `input` element to reset it. You just need to change the `value` prop of the `<Form.Autosuggest` to be an empty string. Wouldn't that clear it?

**Comment by rpenido** [comment]
It uses the `value` only in the first render to set the `displayValue` internal state ([ref](https://github.com/openedx/paragon/blob/95e72b5b6050651644086b207bc2ca99fa2b75a9/src/Form/FormAutosuggest.jsx)). 
```javascript
const [state, setState] = useState({
    displayValue: value || '',
    errorMessage: '',
    dropDownItems: [],
  });
```
This `displayValue` is the state that define the `value` of the input:
```javascript
 <FormControl
          ...
          value={state.displayValue}
          ...
/>
```

After the first render, the `displayValue` is updated in the `onChange` (when we type something) or when we select an option.

```javascript
    setState(prevState => ({
      ...prevState,
      dropDownItems: [],
      displayValue: clickedValue,
    }));
```

It could not always render the `value` props because the user input data is handled (and rendered) internally in the component.

I tried to clean the input value using `document.querySelector('...').value = ''`, but it was overrided as soon as I interacted with the component.

I'm not seeing an easy solution to improve it here.

**Comment by brian-smith-tcril** [comment]
> It uses the value only in the first render to set the displayValue internal state ([ref](https://github.com/openedx/paragon/blob/95e72b5b6050651644086b207bc2ca99fa2b75a9/src/Form/FormAutosuggest.jsx)).

I have a PR out to rework the Autosuggest component a bit (with breaking changes) here: https://github.com/openedx/paragon/pull/2899

Is there any chance you could try out that version of the component? I'd be more than happy to update that PR to include solutions for any issues it doesn't solve!

> The Form.Autosuggest should have higher priority than the modal it's in. Can you just keep the modal the same size but add a CSS rule so that the dropdown has a higher z-index? I have asked about this on #wg-frontend in case we're missing something.

I brought this up in the Paragon working group meeting today and I will be looking into possible short term (read: small code change) fixes today. Once a short term fix is in place, the plan is to look into how to best integrate z-index hooks into Paragon.

**Comment by bradenmacdonald** [comment]
Thanks @brian-smith-tcril !

**Comment by rpenido** [comment]
Thank you @brian-smith-tcril! I think this will cover our use case.

> I will work to update the PR to address this.

I took a fast look at the code and didn't find any immediate problem. :disappointed: 

From my side, I managed to clear the component using:
```javascript
      const inputRef = document.querySelector('.pgn__form-group input');
      inputRef.value = null;
      const event = new Event('change', { bubbles: true });
      inputRef.dispatchEvent(event);
```

**Comment by brian-smith-tcril** [comment]
I have a PR out for the overflow issue here https://github.com/openedx/paragon/pull/2939

I've made some progress on setting/clearing the value on my other branch, I'll take another pass at it today and share once I get it to a point I'm happy with.

**Comment by brian-smith-tcril** [comment]
I'm happy with the value clearing implementation I have in [this PR](https://github.com/openedx/paragon/pull/2899) now (preview [here](https://deploy-preview-2899--paragon-openedx.netlify.app/components/form/form-autosuggest/)). If you pull in the updated example code from https://github.com/openedx/modular-learning/issues/117#issuecomment-1854327383 (I switched `value={value.userProvidedValue}` to `value={value}`) it should be working as expected now.
